wfl/Controllers/MenuSelectionsController.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Web.Http;
+using System.Threading.Tasks;
+using System.Linq;
+using System.Collections.Generic;
+using Symphono.Wfl.Models;
+using Symphono.Wfl.Database;
+
+namespace Symphono.Wfl.Controllers
+{
+    [RoutePrefix("api/menu-selection")]
+    public class MenuSelectionsController : ApiController
+    {
+        IDBManager DBManager;
+        public MenuSelectionsController(IDBManager dbManager) {
+           this.DBManager = dbManager;
+        }
+
+        [Route("")]
+        [HttpGet]
+        public async Task<IHttpActionResult> GetAsync()
+        {
+            MenuSelectionCollection menuSelectionCollection = new MenuSelectionCollection()
+            {
+                MenuSelections = await DBManager.GetAllEntitiesAsync<MenuSelection>()
+            };
+            return Ok(menuSelectionCollection);
+        }
+
+        [Route("{id}")]
+        [HttpGet]
+        public async Task<IHttpActionResult> GetByIdAsync([FromUri] string id)
+        {
+            return Ok(await DBManager.GetEntityByIdAsync<MenuSelection>(id));
+        }
+
+        [Route("")]
+        [HttpPost]
+        public async Task<IHttpActionResult> CreateMenuSelectionAsync(MenuSelectionDto selection)
+        {
+            IEnumerable<FoodOrder> todaysOrders = await DBManager.GetEntitiesByDateAsync<FoodOrder>(DateTime.Now);
+            if (todaysOrders.Any() == false || string.IsNullOrEmpty(selection?.Description) || string.IsNullOrEmpty(selection?.FoodOrderId) || await DBManager.GetEntityByIdAsync<FoodOrder>(selection.FoodOrderId) == null)
```

Comment:
No, that wouldn't be valid. The story said to check if there are any orders scheduled for today, but it may be more useful to check if the food order referenced is scheduled for today.